### PR TITLE
added loading until the API request resolve

### PR DESCRIPTION
**Pull request by EL-Gazoly**
I have included a loading spinner feature. The loading spinner will be displayed until the request has been fully resolved

ps: for some reason, my pic won't appear probably in the about page I think you are cropping your pics with some tool that worked for you but it does not work with me for some reason

### CODE REVIEW

File: src/pages/opportunities/index.astro [hunk 1]
```diff
@@ -15,7 +15,14 @@ url='/opportunities'>
 <div class="mt-40 mx-4 sm:mx-10 text-[#002838] text-xl font-bold">
     <SectionTitle title="Explore New Career Opportunities" style="mx-4 sm:mx-32 items-start" />
 </div>
-<SearchBar />
+<SearchBar id="searchBar"/>
```
Comment (Mahmoudgalalz) [inline src/pages/opportunities/index.astro:18]: I think you can not use id here, remove it

Comment (EL-Gazoly) [reply src/pages/opportunities/index.astro:18]: Yup I was just trying something out

**Review by Mahmoudgalalz — CHANGES_REQUESTED** [review]
and I want the whole page to be in the reload mode,
image
this is a scratch, so you can understand what I'm saying

File: src/pages/opportunities/index.astro (hunk 1 above)
Comment (Mahmoudgalalz) [reply src/pages/opportunities/index.astro:18]: I was doing the same, but its return null, as it's not an HTML class component :joy:

**Comment by EL-Gazoly** [comment]
See I was going to do that but I thought the search bar is static it won't change so why load it again only the data needed to be loaded depending on the query

**Comment by Mahmoudgalalz** [comment]
> See I was going to do that but I thought the search bar is static it won't change so why load it again only the data needed to be loaded depending on the query

way may consider it, like that, but I prefer that, cuz may he will search before loading, so I'm forcing him to wait

and leave the current loading when he click on the search, we will use it

**Comment by EL-Gazoly** [comment]
Okay, as you like, I will start working on it

File: src/pages/opportunities/index.astro (hunk 1 above)
Comment (EL-Gazoly) [reply src/pages/opportunities/index.astro:18]: Yup  you have to surround it on another div tag for example and call that div

**Review by Mahmoudgalalz — APPROVED** [review]
Perfect, I added some adjustments

**Pull request merged**